ra/1.1/resource-agent-api.md

```diff
@@ -198,90 +251,103 @@ unsupported action the RA shall return an error as defined below.
     error must be returned.
 
     The `stop` request by the RM includes the authorization to bring down the
-    resource even by force as long data integrity is maintained; breaking
+    resource even by force as long as data integrity is maintained. Breaking
     currently active transactions should be avoided, but the request to offline
     the resource has higher priority than this. If this is not possible,
-    the RA shall return an error to allow higher level recovery.
+    the RA shall return an error, to allow higher-level recovery.
 
-    The `stop` action should also perform clean-ups of artifacts like leftover
-    shared memory segments, semaphores, IPC message queues, lock files etc.
+    The `stop` action should also clean up any artifacts such as leftover
+    shared memory segments, semaphores, IPC message queues, lock files, etc.
 
     `stop` must succeed if the resource is already stopped.
 
     `stop` must return an error if the resource is not fully stopped.
-  
+
 - `monitor`
 
-    Mandatory.
+    This must check the current status of the resource instance. The
+    thoroughness of the check may optionally be influenced by **Check Levels**.
+    Service must remain available during the monitor, regardless of check
+    level.
 
-    Checks and returns the current status of the resource instance. The
-    thoroughness of the check is further influenced by the weight of the
-    check, which is further explained under **Action specific extensions**..
+- `meta-data`
 
-    It is accepted practice to have additional instance parameters which
-    are not strictly required to identify the resource instance but are
-    needed to monitor it or customize how intrusive this check is allowed
-    to be.
+    This must display the information described under
+    **Resource Agent Meta-Data** via standard output.
 
-    Note that `monitor` shall also return a well defined error code (see
-    below) for stopped instances, ie before `start` has ever been
-    invoked.
-  
-- `recover`
+#### Optional Actions
+
+- `demote`
 
-    Optional.
+    If the resource agent supports roles, this action must put an active
+    resource in the unpromoted role.
+
+- `notify`
+
+    If the resource requires special coordination when multiple instances are
+    run simultaneously in the cluster, the resource agent should support this
+    action, which should perform such coordination.
+
+    When the RA supports this action, RMs should call the action for all active
+    instances of this particular resource in the cluster before and after any
+    demote, promote, start, or stop action performed on any instance of it.
+
+    How the RM passes useful information to the RA when performing this action
+    is currently left to the RM and RA, but may be formalized in a future
+    version of this standard.
+
+- `promote`
+
+    If the resource agent supports roles, this action must put an active
+    resource in the promoted role.
```

Comment:
Should we explicitly state that demote/promote must succeed if the resource is already unpromoted/promoted, as we've done for start and stop actions?

It's covered under the "must be idempotent" clause, regardless.

Reply:
I think I'd rather drop the redundant references in start/stop, but I'll leave those alone at this stage